docs/quickstarts/nextjs.mdx

```diff
@@ -118,55 +125,88 @@ The [`<ClerkProvider>`](/docs/components/clerk-provider) component provides acti
   </Tab>
 </Tabs>
 
-### Require authentication to access your app
+### Add authentication to your app
 
-Now that Clerk is installed and mounted in your application, you can decide which pages are public and which should require authentication to access.
+Now that Clerk is installed and mounted in your application, you can decide which pages are public and which should require authentication to access:
 
-Create a `middleware.ts` file. If your application uses the `src/` directory, your `middleware.ts` file should be placed inside the `src/` folder. If you are not using a `src/` directory, place the `middleware.ts` file in your root directory alongside `.env.local`.
+1. Create a `middleware.ts` file at the root of your project, or in your `src/` directory if you have one.
+2. In your `middleware.ts`, export Clerk's [`authMiddleware`](https://clerk.com/docs/references/nextjs/auth-middleware) helper:
+    ```tsx filename="middleware.ts" {1,3}
+    import { authMiddleware } from "@clerk/nextjs";
 
-Copy and paste the following code into your `middleware.ts`. With this, your entire application is protected.
+    export default authMiddleware({});
+    ```
+3. With `authMiddleware()`, authentication will be enabled by default on all routes that your Next.js middleware runs on. You can specify valid routes using Next.js's [`matcher`](https://nextjs.org/docs/app/building-your-application/routing/middleware#matcher). Add the following code to your `middleware.ts` to protect your entire application:
+    ```tsx filename="middleware.ts" {8}
+    import { authMiddleware } from "@clerk/nextjs";
 
-```tsx filename="middleware.ts"
-import { authMiddleware } from "@clerk/nextjs";
+    export default authMiddleware({});
 
-// This example protects all routes including api/trpc routes
-// Please edit this to allow other routes to be public as needed.
-// See https://clerk.com/docs/references/nextjs/auth-middleware for more information about configuring your Middleware
-export default authMiddleware({});
+    export const config = {
+      // Protects all routes, including api/trpc.
+      // Edit this to allow other routes to be public.

```

Comment:
With `authMiddleware()` the recommendation is to use `publicRoutes: []` to make routes public. `authMiddleware()` has both `ignoredRoutes: []` and `publicRoutes: []`. Removing a route from the Matcher would make it ignored and not public -- there would be no auth information available for that route. This will change with v5 soon, where the new Middleware `clerkMiddleware()` is opt-in.

Also, while the old message is a little unwieldy with the link in the comment, adding that link resulted in a solid decrease in middleware related questions. I would strongly suggest keeping some version of the following, with the link, in the sample code (and above `authMiddleware()` since it pertains to that.

```
// This example protects all routes including api/trpc routes
// Please edit this to allow other routes to be public as needed.
// See https://clerk.com/docs/references/nextjs/auth-middleware for more information about configuring your Middleware
```

Reply:
Thanks for pointing this out! I'll update the text and add the link back 👍